### ISSUE TITLE
Does not work with 'set -o nounset' because of undefined variables usage

### ISSUE THREAD

**Issue by cyril-bouthors** [nojhan/liquidprompt#354]
Liquidprompt does not work when enabling the BASH 'nounset' feature, each prompt shows the three following warnings:

```
09:08:57 θ81° [cyb@home:~] $ set -o nounset
-bash: ret: unbound variable
-bash: LP_TIME_ANALOG: unbound variable
-bash: http_proxy: unbound variable
09:09:02 θ81° [cyb@home:~] $ 
```

Would it be possible to make sure variables are defined before using them?

**Comment by dolmen** [comment]
For reference (at least for myself), here is the relevant part of the `bash` manual about `nounset`:

> Treat unset variables and parameters other than the special parameters "@" and "*" as an error when  performing parameter  expansion.   If  expansion is attempted on an unset variable or parameter, the shell prints  an  error message,  and, if not interactive, exits with a non-zero status.

About your question, I think it would be a big effort (if even possible) to make liquidprompt clean for `nounset`, and I don't intend to spend my own time resources on this effort. But patches are welcome!

Note that on Ubuntu, filename completion seems to have issues working with `nounset`, so just fixing liquidprompt would not be enough.

**Comment by dolmen** [comment]
Related: #360 (fixed by #361).

**Comment by dolmen** [comment]
`bash-completion` (which is something most liquidprompt+bash users will use, as I do myself) is not yet `nounset`-safe. This is on [`bash-completion` roadmap](https://wiki.debian.org/Teams/BashCompletion/Proposals/Roadmap) for 3.0.

Anyway, I have some work in progress in the `fix/usage-of-undef-variables` branch...

**Comment by dolmen** [comment]
I've fixed all cases I found so far with a8114dd9550e2e7fd33b93eb7885de08b3e64933.

**Comment by Rycieos** [comment]
I have both manually inspected the source for `set -u` errors as well as tested thoroughly. This is now greatly improved in [v2.0.0-beta.1](https://github.com/nojhan/liquidprompt/releases/tag/v2.0.0-beta.1).

However, as dolmen noted in https://github.com/nojhan/liquidprompt/issues/354#issuecomment-103235597, `bash-completion` _still_ does not support `set -u`, so I can't run `set -u` on my daily driver shells. There might still be some edge case I failed to catch. Please open a new issue if you find one.
